integration/tests/create_channel.test.ts

```diff
@@ -0,0 +1,209 @@
+
+        const postBalance = await blockUtils.getErc20Balance(
+            CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS,
+            identity.walletAddress
+        );
+
+        expect(postBalance.rawBalance).toBe(prevBalance.rawBalance - depositAmount);
+    });
+
+    it('should approve, deposit, and create channel in several operation', async () => {
+        const prevBalance = await blockUtils.getErc20Balance(
+            CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS,
+            identity.walletAddress
+        );
+
+        const approveTxHash = await client.approveTokens(CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS, depositAmount);
+
+        const approveReceipt = await blockUtils.waitForTransaction(approveTxHash);
+        expect(approveReceipt).toBeDefined();
+
+        const depositTxHash = await client.deposit(CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS, depositAmount);
+        expect(depositTxHash).toBeDefined();
+
+        const depositReceipt = await blockUtils.waitForTransaction(depositTxHash);
+        expect(depositReceipt).toBeDefined();
+
+        const postBalance = await blockUtils.getErc20Balance(
+            CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS,
+            identity.walletAddress
+        );
+
+        expect(postBalance.rawBalance).toBe(prevBalance.rawBalance - depositAmount);
+
+        const openChannelPromise = ws.waitForMessage(getChannelUpdatePredicateWithStatus(RPCChannelStatus.Open), undefined, 5000);
+
+        const {
+            txHash: createChannelTxHash,
+            channelId,
+            initialState,
+        } = await client.createChannel(CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS, {
+            initialAllocationAmounts: [depositAmount, BigInt(0)],
+            stateData: '0x',
+        });
+
+        expect(channelId).toBeDefined();
+        expect(initialState).toBeDefined();
+        expect(createChannelTxHash).toBeDefined();
+
+        const createChannelReceipt = await blockUtils.waitForTransaction(createChannelTxHash);
+        expect(createChannelReceipt).toBeDefined();
+
+        const openResponse = await openChannelPromise;
+        expect(openResponse).toBeDefined();
+
+        const openParsedResponse = parseRPCResponse(openResponse) as ChannelsUpdateRPCResponse;
+        const responseChannel = openParsedResponse.params[0];
+
+        expect(responseChannel.adjudicator).toBe(CONFIG.ADDRESSES.DUMMY_ADJUDICATOR_ADDRESS);
+        expect(responseChannel.amount).toBe(Number(depositAmount));
+        expect(responseChannel.chain_id).toBe(CONFIG.CHAIN_ID);
+        expect(responseChannel.challenge).toBe(CONFIG.DEFAULT_CHALLENGE_TIMEOUT);
+        expect(responseChannel.channel_id).toBe(channelId);
+        expect(responseChannel.participant).toBe(identity.sessionAddress);
+        expect(responseChannel.status).toBe(RPCChannelStatus.Open);
+        expect(responseChannel.token).toBe(CONFIG.ADDRESSES.USDC_TOKEN_ADDRESS);
+    });
+
+    // TODO: find a way to know that broker decided not to join channel
```

Comment:
As Broker joins only channels with configuration he supports, and he does that by listening to blockchain events, he has no way to signal the user that he denies joining the channel. However, we can introduce a "channel join timeout", upon reaching which it can be concluded that Broker either decided not to join or is down.